src/ol/layer/group.js

```diff
@@ -221,6 +221,10 @@ ol.layer.Group.prototype.getLayerStatesArray = function(opt_states) {
         layerState.extent = ownLayerState.extent;
       }
     }
+    if (layerState.layer.getZIndex() === undefined) {
+      layerState.zIndex = ownLayerState.zIndex;
+      layerState.layer.setZIndex(ownLayerState.zIndex);
```

Comment:
I was debating this myself but figured since we are inheriting the `ol.layer.Group`'s zIndex we would want to carry that over to the properties.  I'll remove it if you don't think it makes sense.